src/Common/tests/gtest_async_loader.cpp

```diff
@@ -643,6 +643,72 @@ TEST(AsyncLoader, CustomDependencyFailure)
     ASSERT_EQ(good_count.load(), 3);
 }
 
+TEST(AsyncLoader, WaitersLimit)
+{
+    AsyncLoaderTest t(16);
+
+    std::atomic<int> waiters_total{0};
+    int waiters_limit = 5;
+    auto waiters_inc = [&] (const LoadJobPtr &) {
+        int value = waiters_total.load();
```

Comment:
Same, loading value inside the `while` loop? Otherwise, it won't store the most recent value in the case if `compare_exchange_strong` has failed.